src/overlays/actors/ovl_En_Fish/z_en_fish.c

```diff
@@ -47,73 +63,709 @@ static ColliderJntSphInit D_80A17040 = {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EV_FISH_LEAP);
+        } else {
+            D_80A17018 = 0.0f;
+        }
+    } else {
+        D_80A17018 -= 0.4f;
+    }
+
+    this->skelAnime.playSpeed = ((sp24 + sp20) * 0.5f) + 2.0f;
+    SkelAnime_Update(&this->skelAnime);
+}
+
+void EnFish_Cutscene_WiggleFlyingThroughAir(EnFish* this, GlobalContext* globalCtx) {
+    s32 pad;
+    f32 sp28 = Math_SinS(this->slowPhase);
+    f32 sp24 = Math_SinS(this->fastPhase);
+
+    this->actor.shape.rot.x -= 500;
+    this->actor.shape.rot.z += 100;
+    Math_StepToF(&D_80A17014, 0.0f, 1.0f);
+    this->skelAnime.playSpeed = ((sp28 + sp24) * 0.5f) + 2.0f;
+    SkelAnime_Update(&this->skelAnime);
+}
+
+void EnFish_UpdateCutscene(EnFish* this, GlobalContext* globalCtx) {
+    s32 pad;
+    s32 pad2;
+    CsCmdActorAction* csAction = globalCtx->csCtx.npcActions[1];
+    Vec3f startPos;
+    Vec3f endPos;
+    f32 progress;
+    s32 bgId;
+
+    if (csAction == NULL) {
+        // Warning : DEMO ended without dousa (action) 3 termination being called
+        osSyncPrintf("Warning : dousa 3 消滅 が呼ばれずにデモが終了した(%s %d)(arg_data 0x%04x)\n", "../z_en_sakana.c",
+                     1169, this->actor.params);
+        EnFish_ClearCutsceneData(this);
+        Actor_Kill(&this->actor);
+        return;
+    }
+
+    this->slowPhase += 0x111;
+    this->fastPhase += 0x500;
+
+    switch (csAction->action) {
+        case 1:
+            EnFish_Cutscene_FlopOnGround(this, globalCtx);
+            break;
+        case 2:
+            EnFish_Cutscene_WiggleFlyingThroughAir(this, globalCtx);
+            break;
+        case 3:
+            // DEMO fish termination
+            osSyncPrintf("デモ魚消滅\n");
+            EnFish_ClearCutsceneData(this);
+            Actor_Kill(&this->actor);
+            return;
+        default:
+            // Improper DEMO action
+            osSyncPrintf("不正なデモ動作(%s %d)(arg_data 0x%04x)\n", "../z_en_sakana.c", 1200, this->actor.params);
+            break;
+    }
+
+    startPos.x = csAction->startPos.x;
+    startPos.y = csAction->startPos.y;
+    startPos.z = csAction->startPos.z;
+    endPos.x = csAction->endPos.x;
+    endPos.y = csAction->endPos.y;
+    endPos.z = csAction->endPos.z;
+
+    progress = func_8006F93C(csAction->endFrame, csAction->startFrame, globalCtx->csCtx.frames);
+
+    this->actor.world.pos.x = ((endPos.x - startPos.x) * progress) + startPos.x;
+    this->actor.world.pos.y = ((endPos.y - startPos.y) * progress) + startPos.y + D_80A17014;
+    this->actor.world.pos.z = ((endPos.z - startPos.z) * progress) + startPos.z;
+
+    this->actor.floorHeight = BgCheck_EntityRaycastFloor4(&globalCtx->colCtx, &this->actor.floorPoly, &bgId,
+                                                          &this->actor, &this->actor.world.pos);
+}
+
+// Update functions and Draw
+
+void EnFish_OrdinaryUpdate(EnFish* this, GlobalContext* globalCtx) {
+    if (this->timer > 0) {
```

Comment:
not a fan of "ordinary update", but have no suggestions for now

Reply:
Yeah, the problem with this function is it's very much "not any of the special updates", since Update itself is basically a wrapper to choose which sub-update to use. And something like UpdateNormal or NormalUpdate sounds like it's to do with normals or something